src/refactor_nrepl/ns/slam/hound/regrow.clj

```diff
@@ -20,6 +28,22 @@
          v#))
      (do ~@body)))
 
+(defmacro cache-with-dirty-tracking
+  "The function to be cached, f, should have two signatures. A zero-operand
+  signature which computes the result for all namespaces, and a two-operand
+  version which takes the previously computed result and a list of dirty
+  namespaces, and amends the result."
```

Comment:
I've just never heard anyone talk about 'amending' a datastructure :) I think 'updates' is better, yeah.